src/pandablocks_ioc/_pvi.py

```diff
@@ -61,36 +56,28 @@ def add_pvi_info(
 ) -> None:
     """Create the most common forms of the `PviInfo` structure"""
     component: Component
-    writeable: bool = record_creation_func in OUT_RECORD_FUNCTIONS
-    useComboBox: bool = record_creation_func == builder.mbbOut
-
-    if record_creation_func == builder.Action:
-        if record_name == "PCAP:ARM":
-            component = SignalRW(
-                record_name,
-                record_name,
-                widget=ButtonPanel(actions=dict(Arm=1, Disarm=0)),
-                read_widget=LED(),
-            )
-            access = "rw"
 
-        else:
-            component = SignalX(record_name, record_name, value="")
```

Comment:
The following PVs are made into `SignalX` here:
`SFP2_SYNC_IN:SYNC_RESET`
`SFP3_SYNC_IN:SYNC_RESET`
`SRGATE1:FORCE_SET`
`SRGATE1:FORCE_RST`
`SRGATE2:FORCE_SET`
`SRGATE2:FORCE_RST`
`SRGATE3:FORCE_SET`
`SRGATE3:FORCE_RST`
`SRGATE4:FORCE_SET`
`SRGATE4:FORCE_RST`

I've tried to streamline the code so that widgets are received for a given builder function from a dictionary. Why don't we want these having the same widget as `PCAP:ARM`? - I suppose because not all are toggle-able?

Reply:
I recall deciding against that particular approach as it would end up scattering widget definitions everywhere that we create records. This would break the current encapsulation we have with PVI, which is (nearly?) entirely contained in this one file.

But I can see the issue you're hitting. Perhaps there's a half-way house to be found? Something like passing an extra parameter to this function, perhaps an `enum`, specifying what kind of widget(s) should be made? That gives more control to the caller, as they know how they want their records to be displayed, without having to explicitly add PVI-specific code everywhere. This approach may also neaten up some of the increased complexity in this function by allowing us to do away with many of the `if-else` statements.

Reply:
I like this idea, I'll keep the enum in `_types.py`.